kimchi/src/verifier.rs

```diff
@@ -480,6 +480,42 @@ where
     }
 }
 
+fn check_proof_evals_len<G>(proof: &ProverProof<G>)
+where
+    G: KimchiCurve,
+    G::BaseField: PrimeField,
+{
+    let ProofEvaluations {
+        w,
+        z,
+        s,
+        coefficients,
+        lookup,
+        generic_selector,
+        poseidon_selector,
+    } = &proof.evals;
+
+    let check_eval_len = |eval: &PointEvaluations<Vec<_>>| {
+        assert!(eval.zeta.len() == 1);
+        assert!(eval.zeta_omega.len() == 1);
+    };
+
+    w.iter().for_each(check_eval_len);
+    s.iter().for_each(check_eval_len);
+    coefficients.iter().for_each(check_eval_len);
+    if let Some(lookup) = lookup {
+        lookup.sorted.iter().for_each(check_eval_len);
+        check_eval_len(&lookup.aggreg);
+        check_eval_len(&lookup.table);
+        if let Some(runtime) = &lookup.runtime {
+            check_eval_len(runtime);
+        }
+    }
+    check_eval_len(z);
```

Comment:
nit: it might be nicer to have these checks follow the order of the fields in the struct

Reply:
Pls check https://github.com/o1-labs/proof-systems/pull/989/commits/d05bdd1287a6d6d2f767eacbfde2163aa552a250